tensorflow/core/kernels/segment_reduction_ops.h

```diff
@@ -66,42 +68,76 @@ struct UnsortedSegmentFunctor {
 };
 
 #if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
-// reduction functors for the gpu
+
+// Atomic reduction functors for the gpu.
 template <typename T>
-struct SumOpGpu {
+struct SumAtomicOpGpu {
   EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE void operator()(T* dest,
                                                         const T& value) {
     GpuAtomicAdd(dest, value);
   }
 };
 
 template <typename T>
-struct ProdOpGpu {
+struct ProdAtomicOpGpu {
   EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE void operator()(T* dest,
                                                         const T& value) {
     GpuAtomicMul(dest, value);
   }
 };
 
 template <typename T>
-struct MaxOpGpu {
+struct MaxAtomicOpGpu {
   EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE void operator()(T* dest,
                                                         const T& value) {
     GpuAtomicMax(dest, value);
   }
 };
 
 template <typename T>
-struct MinOpGpu {
+struct MinAtomicOpGpu {
   EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE void operator()(T* dest,
                                                         const T& value) {
     GpuAtomicMin(dest, value);
   }
 };
 
+// Reduction functors for the gpu.
+template <typename T>
+struct SumOpGpu {
```

Comment:
Can you rename this to `NonAtomicSumOpGpu`?  That will make it obvious that this PR did not accidentally make an atomic operation non-atomic.

Reply:
It indeed makes sense, and I have modified as you said.